chromadb/api/shared_system_client.py

```diff
@@ -94,3 +94,51 @@ def _system(self) -> System:
     def _submit_client_start_event(self) -> None:
         telemetry_client = self._system.instance(ProductTelemetryClient)
         telemetry_client.capture(ClientStartEvent())
+
+    @staticmethod
+    def get_chroma_cloud_api_key_from_clients() -> Optional[str]:
+        """
+        try to extract api key from existing client instaces by checking httpx session headers
+        if available.
+
+        Requirements to pull api key:
+        - must be a FastAPI instance (ignore RustBindingsAPI and SegmentAPI)
+        - must have a "api.trychroma.com" in the _api_url (ignore local/self-hosted instances)
+        - must have "x-chroma-token" or "X-Chroma-Token" in the headers
+
+        Returns:
+            The first api key found, or None if no client instances have api keys set.
+        """
+        # check FastAPI instance session headers bc this is where both cloudclient and httpclient paths converge
+        systems_snapshot = list(SharedSystemClient._identifier_to_system.values())
```

Comment:
I think the behavior of grabbing the first api key in this list is confusing and it should be instead that if multiple api keys are found we log the one we are using OR error. To log the api key we can log the first 5 characters of it etc. The idea being that we do something well defined instead of silently grabbing the first one.